internal/server/data/migrations.go

```diff
@@ -1258,3 +1259,53 @@ func removeSettingsPasswordPolicy() *migrator.Migration {
 		},
 	}
 }
+
+func addAccessKeyIssuedForKind() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2023-01-17T11:00",
+		Migrate: func(tx migrator.DB) error {
+			if !migrator.HasColumn(tx, "access_keys", "issued_for") {
+				return nil
+			}
+			if _, err := tx.Exec(`DROP INDEX IF EXISTS idx_access_keys_issued_for_name`); err != nil {
+				return fmt.Errorf("drop access key issued_for index: %w", err)
+			}
+			stmt := `
+				ALTER TABLE access_keys RENAME COLUMN issued_for TO issued_for_id;
+				ALTER TABLE access_keys ADD COLUMN IF NOT EXISTS issued_for_kind text DEFAULT 'user';
+			`
+			if _, err := tx.Exec(stmt); err != nil {
+				return fmt.Errorf("add access key issued_for_kind and issued_for_id: %w", err)
+			}
+			if _, err := tx.Exec(`UPDATE access_keys SET issued_for_kind = 'provider' WHERE issued_for_id = provider_id;`); err != nil {
+				return fmt.Errorf("set access key issued_for_kind provider: %w", err)
+			}
+
+			stmt = `SELECT id FROM identities WHERE deleted_at IS NULL AND name = 'connector';`
+			rows, err := tx.Query(stmt)
+			if err != nil {
+				return err
+			}
+			connectorIDs, err := scanRows(rows, func(id *uid.ID) []any {
+				return []any{id}
+			})
+			if err != nil {
+				return fmt.Errorf("read connector identity rows: %w", err)
+			}
+			if len(connectorIDs) > 0 {
+				query := querybuilder.New(`UPDATE access_keys`)
+				query.B(`SET issued_for_kind = 'organization'`)
```

Comment:
Does this impact how connector keys are presented to the user?

Reply:
At the moment it is still presented by "user=connector" and not changed, but in a future change we can set a filter based on "kind=organization" instead.